src/enclave_proc/mod.rs

```diff
@@ -358,41 +358,55 @@ fn handle_command(
         }
 
         EnclaveProcessCommandType::Describe => {
+            let describe_args = connection.read::<DescribeArgs>().map_err(|e| {
+                e.add_subaction("Failed to get describe arguments".to_string())
+                    .set_action("Describe Enclave".to_string())
+            })?;
             connection.write_u64(MSG_ENCLAVE_CONFIRM).map_err(|e| {
                 e.add_subaction("Failed to write confirmation".to_string())
                     .set_action("Describe Enclaves".to_string())
             })?;
 
             // Evaluate thread result at first describe, then set thread to None.
             if pcr_thread.is_some() {
-                enclave_manager
```

Comment:
Moved to `fetch_describe_result` function.

But generally I would go further and refactor the approach of `DescribeThread` logic: it is to implicit and prone to bugs. It would be better to keep the thread logic within `EnclaveManager` itself rather then setting the measurements and metadata result from the outside.

But probably such a change not for now, this MR is already too huge.